### PR TITLE
Release 0.4.2

### PR DESCRIPTION
**Pull request by Bodigrim**

### CODE REVIEW

**Comment by cartazio** [comment]
@you're on the uploader lists :)

File: Changes [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+    Add basic functions on Gaussian integers.
+    Add Moebius mu-function.
+
+    Forbid non-positive moduli in Math.NumberTheory.Moduli.
+
+    Fix out-of-bounds error in Math.NumberTheory.Primes.Heap, Math.NumberTheory.Primes.Sieve and Math.NumberTheory.MoebiusInversion.
+    Fix 32-bit build.
+    Fix binaryGCD on negative numbers.
+    Fix highestPower (various issues).
+
```
Comment (cartazio) [inline Changes]: needs to say "this work was by Andrew Lelechenko, who is awesome and meticulous" or should I just be sure to state that in the ANN we post on cafe and reddit :)

File: arithmoi.cabal [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name                : arithmoi
-version             : 0.4.1.3
+version             : 0.4.2.0
 cabal-version       : >= 1.10
 author              : Daniel Fischer
```
Comment (cartazio) [inline arithmoi.cabal]: should we drop the -fllvm flag? I'm pretty sure motivated users can figure out --ghc-options="-fllvm" themselves! (do we have benchmarks to validate exposing it at our level at all?)

**Comment by Bodigrim** [comment]
I have never used `llvm` flag. Removed.

**Comment by cartazio** [comment]
Oh, what's the hpc coverage report if we do a cabal configure
--enable-tests --enable-coverage ; cabal test ?
Probably good to see which code paths are / aren't tested before we do the
final cut!

I'm away from keyboard but I can have a go later tonight when I'm back by
my computer this evening. Cause I think having a good idea of what code
paths haven't been excericsed before we do the release cut might be a good
idea.

It's late in your time zone, I'll have a look this evening. !

**Comment by Bodigrim** [comment]
There are two untested areas: `Math.NumberTheory.Primes.Factorisation` and `Math.NumberTheory.Primes.Testing.Probabilistic`. There were no changes in both (except new `moebius` function #17) since previous release, so we can be sure that there is no regression.

**Comment by cartazio** [comment]
Cool! I'm still amidst family travels till tomorrow evening. But that sort
of test coverage  improvement in the test suite is worth mentioning maybe?
Granted most users won't care :)
On May 29, 2016 3:10 PM, "Bodigrim" <notifications@github.com> wrote:

There are two untested areas: Math.NumberTheory.Primes.Factorisation and
Math.NumberTheory.Primes.Testing.Probabilistic. There were no changes in
both (except new moebius function #17
<https://github.com/cartazio/arithmoi/pull/17>) since previous release, so
we can be sure that there is no regression.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/cartazio/arithmoi/pull/29#issuecomment-222385729>, or mute
the thread
<https://github.com/notifications/unsubscribe/AAAQwqcaog36AVSKoLy1Dh_LE6T5CWZrks5qGg6_gaJpZM4Im-zT>
.

**Comment by cartazio** [comment]
If you want to cut the release go crazy. Lemme know if you want any help
doing the Haskell Reddit or cafe mailing list Announce but all looks good
to me

**Comment by Bodigrim** [comment]
I have uploaded 0.4.2.0 on Hackage. 
It would be incredibly cool, if you get time to write Reddit/cafe Announce :)

**Comment by cartazio** [comment]
Its in my calendar for this evening. Please mock me via gchat and email if
it doesnt happen in the next 12 hours (and hoping no one scoops it :) )

**Pull request merged**